src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/LocationAttribute.kt

```diff
@@ -0,0 +1,97 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity
+
+import jakarta.persistence.CascadeType
+import jakarta.persistence.Entity
+import jakarta.persistence.EnumType
+import jakarta.persistence.Enumerated
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.OneToMany
+import jakarta.persistence.OneToOne
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.UUID
+
+@Entity
+@Table(name = "location_attribute")
+data class LocationAttribute(
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  val locationAttributeId: Long = 0,
+
+  val dpsLocationId: UUID,
+
+  @OneToOne
+  @JoinColumn(name = "prison_id")
+  val prison: Prison,
+
+  @Enumerated(EnumType.STRING)
+  val locationStatus: LocationStatus = LocationStatus.ACTIVE,
+
+  val statusMessage: String? = null,
+
+  val expectedActiveDate: LocalDate? = null,
+
+  @Enumerated(EnumType.STRING)
+  val locationUsage: LocationUsage,
+
+  val allowedParties: String? = null,
+
+  val prisonVideoUrl: String? = null,
+
+  val notes: String? = null,
+
+  val createdBy: String,
+
+  val createdTime: LocalDateTime = LocalDateTime.now(),
+) {
+  var amendedBy: String? = null
+    private set
+
+  var amendedTime: LocalDateTime? = null
+    private set
+
+  @OneToMany(mappedBy = "locationAttribute", fetch = FetchType.LAZY, cascade = [CascadeType.ALL], orphanRemoval = true)
+  var locationSchedule: MutableList<LocationSchedule> = mutableListOf()
+    private set
+
+  fun setLocationSchedule(schedules: MutableList<LocationSchedule>) {
+    this.locationSchedule = schedules
```

Comment:
Perhaps this should fail if the location usage is not schedule? e.g

```
    require(locationUsage == LocationUsage.SCHEDULE) {
      "blah blah" 
    }
```

Reply:
Done.